src/Language/Mulang/JSCompiler.hs

```diff
@@ -0,0 +1,358 @@
+    return [text| function(){ $cases }($value) |]
+
+    where compileCase (_target, _body) = do
+            target <- compile _target
+            body   <- compile _body
+            return [text| if(arguments[0] === $target) { return $body } |]
+
+  -- Generates a JS function for the class. The superclass assignation is lazily applied to compensate for the lack of
+  -- hoisting.
+  compile (Class _name _superclassName _body) = do
+    let name = pack _name
+    let superclassName = pack $ fromMaybe "MuObject" _superclassName
+    body <- compileBody _body
+    return [text| function $name() {
+                    this.constructor.prototype = Object.create($superclassName.prototype);
+                    $superclassName.call(this);
+                    $body
+                  }
+           |]
+    where compileBody :: Expression -> Maybe Text
+          compileBody (Sequence expressions) = compileAll expressions ";\n"
+          compileBody MuNull                 = Just [text| |]
+          compileBody expression             = compile expression
+
+  -- Generates a JS new of an instance of an anonymous class.
+  -- I'm not sure how object's superclass is supposed to be represented, but Include of classes should work.
+  compile (MuObject _body) = do
+    singletonClass <- compile $ Class "" Nothing _body
+    return[text| new $singletonClass() |]
+
+  -- Generates a JS variable assignation pointing to an anonymous instance.
+  compile (Object _name _body) = do
+    let name = pack _name
+    object <- compile $ MuObject _body
+    return [text| var $name = $object |]
+
+  -- Generates a JS assignation statement that overrides the context's constructor's prototype slot named as the method
+  -- with a new function for the method.
+  -- By-name access is used in order to support invalid JS selectors.
+  -- The generated code is meant to be evaluated in the context of a container "class" function.
+  compile (Method _name _equations) = do
+    let name = pack _name
+    body <- compile $ Function "" _equations
+    return [text| this.constructor.prototype['${name}'] = $body |]
+
+  compile (EqualMethod _equations) = compile $ Method "===" _equations
+  compile (HashMethod _equations) = compile $ Method "hash" _equations
+
+  -- Generates a JS assignation statement that overrides the context's slot named as the attribute.
+  -- By-name access is used in order to support invalid JS selectors.
+  -- The generated code is meant to be evaluated in the context of a container "class" function.
+  compile (Attribute _name _value) = do
+    let name = pack _name
+    value <- compile _value
+    return [text| this['${name}'] = $value |]
+
+  -- Generates a JS assignation statement that replaces the current context's constructor's prototype with a copy of
+  -- itself merged with the mixin's prototype.
+  -- It also calls the mixin initialization on the new instance.
+  compile (Include _name) = do
```

Comment:
I recommend not using `_identifiers`, since it means _a non interesting value but that still has a name_